Remade the seat generation and added tests for it

PurtucRobert opened the pull request:
- Isolated the seat generation inside a function
- Creation of seats in a hall is triggered using signals
- Added unit tests for seat generation function

PurtucRobert committed:
- Isolated the seat generation inside a function
- Creation of seats in a hall is triggered using signals
- Added unit tests for seat generation function

PurtucRobert commented on cinemy/cinema/apps.py:
```diff
@@ -2,5 +2,9 @@
+
+    def ready(self):
+
+        from cinema.signals import handlers
```
Django specific import

maxinasa commented on cinemy/cinema/signals/handlers.py:
```diff
@@ -0,0 +1,11 @@
+def hall_creation_handler(instance, **kwargs):
+    generated_seats = generate_seats(instance.rows, instance.seats_per_row)
+    for i in range(len(generated_seats)):
+        generated_seats[i].hall = instance
```
Not 100% sure, but see if this might help, instead of the hall assignment for each object:
https://docs.djangoproject.com/en/4.0/ref/models/relations/#django.db.models.fields.related.RelatedManager.add

PurtucRobert replied:
Done

PurtucRobert committed:
- Made the hall field not required
- Added the related name to the seat model
- Refactored the hall_creation_handler
